stock_picking_dispatch/models/picking_dispatch.py

```diff
@@ -174,60 +162,18 @@ def action_cancel(self):
 
     @api.multi
     def assert_start_ok(self):
-        now = datetime.now().date()
-        for obj in self:
-            date = datetime.strptime(obj.date, '%Y-%m-%d').date()
-            if now < date:
-                raise except_osv(
-                    _('Error'),
-                    _('This dispatch cannot be processed until %s') % obj.date)
-
-    def action_assign_moves(self, cr, uid, ids, context=None):
-        move_obj = self.pool['stock.move']
-        location_obj = self.pool['stock.location']
-
-        move_ids = move_obj.search(cr, uid, [
-            ('dispatch_id', 'in', ids),
-            ('state', 'in', ('confirmed', 'waiting')),
-        ], context=context)
-
-        cr.execute("""
-            SELECT
-                product_id,
-                location_id,
-                SUM(product_qty)
-            FROM stock_move
-            WHERE
-                dispatch_id in %s
-                AND state in ('confirmed', 'waiting')
-            GROUP BY product_id, location_id
-            """, (tuple(ids), ))
-        groups = cr.fetchall()
-
-        for product_id, location_id, quantity in groups:
-            # Try to reserve all moves with the same product and location
-            # together
-            res = location_obj._product_reserve(
-                cr,
-                uid,
-                [location_id],
-                product_id,
-                quantity,
-                lock=True,
-                context=context
-            )
-
-            if res and len(res) == 1:
-                move_obj.write(cr, uid, move_ids, {'state': 'assigned',
-                                                   'location_id': res[0][1]},
-                               context=context)
-            else:
-                # We could not reserve all the moves together, or the
-                # reservation is split among many sublocations.
-                # If that's the case, fall back to reserving one move at a time
-                move_obj.action_assign(cr, uid, move_ids, context=context)
+        today = date.today()
```

Comment:
Use `fields.Date.today()` instead or `fields.Date.from_string(dispatch.date)` to avoid using `strptime`